### ISSUE TITLE
Rethink page status option nomenclature

### ISSUE THREAD

**Issue by heyallan** [getkirby/panel#706]
As of now (when creating a page) status option is either: visible/invisible
I suspect it means: draft/published

I think the later naming convention should be used instead, because is more “natural” to the document publishing context.

I know it's not a big deal for power users, but the meaning of words makes a huge difference for clients.

**Comment by JimmyRittenborg** [comment]
I'm not entirely sure about this, because the page actually do become either visible or invisible from the page navigation, but it's still public accessable at its URL ? I don't think a draft would be that. A page i don't want to have show up in the navigation could on the other hand be many things. So I think an draft/unpublished and published option, would work in a different. But it's far from the first time this have been brought up.

**Comment by heyallan** [comment]
@JimmyRittenborg Not sure what you mean in the end really.

But I suspect nor visible/invisible, nor draft/published, are sufficient to convey the implicit meaning of this option (?)

I still think it's worth rethinking.

**Comment by JimmyRittenborg** [comment]
At which point in the end? :)

So you're sure about you just didn't thought the wording was wrong because you didn't knew what the option actually did? Because many mistakes this options for being an published/unpublished option which it's not - to me there's an important difference in misinterpreting wording vs, misinterpreting function, both are equally bad though.

**Comment by lukasbestle** [comment]
I'm pretty sure there was another very similar feature request a while ago, but I can't find it.

The opinion there was that it should be possible to "rename" the state labels in the blueprints, because in the end what the toggle does hugely depends on how the state `visible` is used in the templates.

**Comment by heyallan** [comment]
Exactly!

`visible/invisible` is thought in terms of how template is used.

I don't think the user should have to deal with template logic (that's developer stuff).

The option offered to user should be simpler **to understand**.

For instance, when developing a website for a chef, or artist, I don't think they must be interested in  understanding how a template logic works to know when to toggle the switch. I don't think it should have to be explained at all. Fact is they think in simpler terms like `publish/unpublish`.

**I don't mean limiting the user (nor constraining options) by any means**, just rethinking how to be more explicit in a way that is meaningful to the user.

**My point is:** `visible/invisible` conveys meaning for who's creating templates, not necessarily for who's creating content.

I see room for improvement.

**Comment by texnixe** [comment]
The problem is that visible/invisible is just a flag, on the most basic level it means that the folder name either has a prepended sorting number or not.

What this flag does in the end, depends entirely on the developer, how they use this flag, e.g. they could use it to only include pages with a number in a menu or to only show posts with a number.

For example, I don't want pages like imprint or home to show up in the main menu, so my imprints don't get a number. However, I still want these pages to be visible to the visitors of my website. So in fact these pages are published although they are invisible.

This example is just to show that the renaming of visible/invisible to published/unpublished will not always fit.

What is in fact needed, is an option to completely hide the toggle via a blueprint setting for those cases where the user should not be able to change anything and thereby break the website, and an option to rename the wording to make it fit with the purpose of the flag in a specific project.

**Comment by lukasbestle** [comment]
I completely agree!

**Comment by heyallan** [comment]
I know, I know.

But, look: you set imprint to **invisible**, but visitors load the page and the imprint is **visible** anyway.
See? You set it to invisible, but it is visible 😝

That's an inconsistency, right?

Maybe in your case the toggle really means `page/component` (?) because pages are supposed to be visible, but components are just supposed to be implicit.

**I see `publish/unpublish` might not be the best fit, it's alright.**
But whatever it is, I don't see how `visible/invisible` makes the cut anyway.

The problem might be the **mixed nature** of the toggle.
Because visibility might have different meanings in different scenarios:

- visibility as a menu item (show dropdown or not)
- visibility as a page (publish or unpublish)
- visibility as an `include` in templates (visible or invisible)

Too many possible outcomes for a single binary switch IMO.
**But above all, I don't think the `show-in-menu` option should belong to this particular toggle.**

Not sure if i'm rather getting confused 😛

**Comment by heyallan** [comment]
Yes, I got a bit confused, 😜 but anyways I would like to have some separation between functionalities:

0. publish-page
0. show-in-menu

Just saying :v:

**Comment by texnixe** [comment]
I agree that it might be confusing, because the flag visible/invisible does not mean that the page is or isn't publicly reachable. By default, you can call any page, visible or not, via its URL. Therefore, `3-about` or `about` are both visible to the user if they call the URL http://yourdomain.com/about. This behavior would not change if you just change the nomenclature.

It is therefore important for the developer to understand what the toggle means and what you can do with it. As I explained above, all visibility/invisibility does is add a sorting number. It does not automatically mean that pages without a sorting number are not visible in the menu, because it depends entirely on your menu snippet. Only if you filter by `$pages->visible()` in your menu snippet, you exclude invisible pages.

You can, of course, filter by a field in your text file, e.g. a checkbox field like `include_in_menu` or `publish` to get the same effect (although performance-wise, this might be slower than using the visibility switch, because Kirby needs to read the text files).

And apart from determining whether or not a page should be shown in a menu or visible on the website, there is also the sorting aspect, either by number, date or alphabetically.

You might also want to check out this discussion: https://forum.getkirby.com/t/setting-to-route-all-invisible-pages-to-error-page/1030

**Comment by heyallan** [comment]
I understand how it works, I have no problem implementing it.

My point is the current nomenclature doesn't match the functionality, hence explaining the use of the toggle becomes confusing to some extent (as noted by Bastian [https://forum.getkirby.com/t/setting-to-route…](https://forum.getkirby.com/t/setting-to-route-all-invisible-pages-to-error-page/1030/24?u=heyallan)).

And the fact that it's functional doesn't mean it's doesn't need improvement. 

**My suggestion is the separation of purposes**

Abstraction of components comes into mind, (it's just a binary flag, right?) But, look, it's linked other functions. Sorting by number, date, or alphabetically, is not even binary option.

You see? Too much for a binary flag IMO.
**And as not even nomenclature can bring clarity to this case, I will close this issue.**
